pkg/splunk/enterprise/configuration_test.go

```diff
@@ -278,6 +280,90 @@ func TestSmartStoreConfigDoesNotFailOnIndexerClusterCRForCM(t *testing.T) {
 	}
 }
 
+func TestSmartstoreApplyIndexerClusterCreatesSmartStoreConfigMapOnValidConfig(t *testing.T) {
+	cr := enterprisev1.IndexerCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "idxCluster",
+			Namespace: "test",
+		},
+		Spec: enterprisev1.IndexerClusterSpec{
+			Replicas: 1,
+			SmartStore: enterprisev1.SmartStoreSpec{
+				VolList: []enterprisev1.VolumeSpec{
+					{Name: "msos_s2s3_vol", Endpoint: "https://s3-eu-west-2.amazonaws.com", Path: "testbucket-rs-london"},
+				},
+
+				IndexList: []enterprisev1.IndexSpec{
+					{Name: "salesdata1", VolName: "msos_s2s3_vol"},
+					{Name: "salesdata2", RemotePath: "salesdata2", VolName: "msos_s2s3_vol"},
+					{Name: "salesdata3", RemotePath: "", VolName: "msos_s2s3_vol"},
+				},
+			},
+		},
+	}
+	c := spltest.NewMockClient()
+
+	// Create namespace scoped secret
+	secret, err := ApplyNamespaceScopedSecretObject(c, "test")
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+
+	secret.Data[s3AccessKey] = []byte("34kl*&#$(@)@$)%)#%@#&#$***#$KJL#$KJ#$KL#$")
+	secret.Data[s3SecretKey] = []byte("flkd($*)#$#($#)*&($(%$(*%$%)*($)(*&^&&JKH")
+	err = splctrl.UpdateResource(c, secret)
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+
+	_, err = ApplyIndexerCluster(c, &cr)
+	if err != nil {
+		t.Errorf("ApplyIndexerCluster should not fail for a valid smartstore config, %s", err.Error())
+	}
+}
+
+func TestSmartstoreApplyStandaloneCreatesSmartStoreConfigMapOnValidConfig(t *testing.T) {
```

Comment:
Ditto as above with TestApplyStandAlone?

Reply:
Smartstore config is optional, so, having a separate test is a good choice here.

Reply:
Addressed